tests/fixed-foveation.html

```diff
@@ -0,0 +1,553 @@
+             255,   0, 255, 255 ,    0, 255,   0, 255 ,    0,   0, 255, 255 ,  255,   0,   0, 255 ,
+               0,   0, 255, 255 ,    0, 255,   0, 255 ,  255,   0, 255, 255 ,  255,   0,   0, 255 ,
+
+        ];
+
+        cube.VertexCount = 8;
+        cube.IndexCount = 36;
+
+        var vertexAttribs = new Array(1);
+        vertexAttribs[0] = new Object;
+        vertexAttribs[0].index = vertexPositionLocation;
+        vertexAttribs[0].size = 4;
+        vertexAttribs[0].type = gl.BYTE;
+        vertexAttribs[0].normalize = true;
+        vertexAttribs[0].stride = 0;        // 0 = move forward size * sizeof(type) each iteration to get the next position
+        vertexAttribs[0].offset = 0;        // start at the beginning of the buffer
+
+       /* vertexAttribs[1] = new Object;
+        vertexAttribs[1].index = vertexColorLocation;
+        vertexAttribs[1].size = 4;
+        vertexAttribs[1].type = gl.UNSIGNED_BYTE;
+        vertexAttribs[1].normalize = true;
+        vertexAttribs[1].stride = 0;        // 0 = move forward size * sizeof(type) each iteration to get the next position
+        vertexAttribs[1].offset = 8 * 4;    // skip positions part, offset to color, in bytes
+         */
+        cube.VertexAttribs = vertexAttribs;
+
+        cube.VertexBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ARRAY_BUFFER, cube.VertexBuffer);
+        gl.bufferData(gl.ARRAY_BUFFER, new Uint8Array(cubeVertices), gl.STATIC_DRAW);
+        console.log("cube.VertexBuffer size " + gl.getBufferParameter(gl.ARRAY_BUFFER, gl.BUFFER_SIZE));
+        gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+        const cubeIndices =
+        [
+          0, 2, 1, 2, 0, 3,	// top
+          4, 6, 5, 6, 4, 7,	// bottom
+          2, 6, 7, 7, 1, 2,	// right
+          0, 4, 5, 5, 3, 0,	// left
+          3, 5, 6, 6, 2, 3,	// front
+          0, 1, 7, 7, 4, 0	// back
+        ];
+
+        cube.IndexBuffer = gl.createBuffer();
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, cube.IndexBuffer);
+        gl.bufferData(gl.ELEMENT_ARRAY_BUFFER, new Uint16Array(cubeIndices), gl.STATIC_DRAW);
+        console.log("cube.IndexBuffer size " + gl.getBufferParameter(gl.ELEMENT_ARRAY_BUFFER, gl.BUFFER_SIZE));
+        gl.bindBuffer(gl.ELEMENT_ARRAY_BUFFER, null);
+
+        return cube;
+      }
+
+      function GeometryCreateVAO(gl, geometry) {
+        geometry.VertexArrayObject = gl.createVertexArray();
+        gl.bindVertexArray(geometry.VertexArrayObject);
+        gl.bindVertexArray(null);
+      }
+
+      function SceneCreateVAOs(gl, scene)
+      {
+        if (!scene.CreatedVAOs )
+        {
+          GeometryCreateVAO(gl, scene.Cube);
+
+          // Modify the VAO to use the instance transform attributes.
+          gl.bindVertexArray(scene.Cube.VertexArrayObject);
+          gl.bindBuffer(gl.ARRAY_BUFFER, scene.InstanceTransformBuffer);
+          gl.bindVertexArray(null);
+
+          scene.CreatedVAOs = true;
+        }
+      }
+
+      function createScene(gl)
```

Comment:
As this is a test page and not a "sample" I'm not too concerned about it, but this is pretty different from the structure of most of our pages. From the looks of the comments it seems like it's ported from some Meta sample code? If it's publicly accessible code it would be good to add a comment somewhere indicating what this code is based on with a URL where the original can be found.